internal/management/controller/instance_controller.go

```diff
@@ -1020,23 +1023,26 @@ func (r *InstanceReconciler) reconcilePrimary(ctx context.Context, cluster *apiv
 		return false, err
 	}
 
+	// If I'm not the primary, let's promote myself
 	if !isPrimary {
-		// If I'm not the primary, let's promote myself
-		err := r.promoteAndWait(ctx, cluster)
-		if err != nil {
+		if err := r.promoteAndWait(ctx, cluster); err != nil {
 			return false, err
 		}
 		restarted = true
 	}
 
-	// If it is already the current primary, everything is ok
+	// if the currentPrimary doesn't match the PodName we set the correct value.
 	if cluster.Status.CurrentPrimary != r.instance.PodName {
+		// we log the timestamp information before overriding them. This is useful to know how much time it took.
+		contextLogger.Info("printing timestamp info before setting myself as the current primary")
+		cluster.LogTimestamps(ctx)
+		contextLogger.Info("Setting myself as the current primary")
 		cluster.Status.CurrentPrimary = r.instance.PodName
 		cluster.Status.CurrentPrimaryTimestamp = pkgUtils.GetCurrentTimestamp()
```

Comment:
we could potentially log the timestamp after we patch the resource on the cluster, but that would make the code a bit more dirty